client/node-common/src/command.rs

```diff
@@ -44,3 +51,179 @@ pub fn generate_genesis_block<Block: BlockT>(
+    }
+
+    fn role(&self, is_dev: bool) -> Result<sc_service::Role> {
+        self.base.base.role(is_dev)
+    }
+
+    fn transaction_pool(&self, is_dev: bool) -> Result<sc_service::config::TransactionPoolOptions> {
+        self.base.base.transaction_pool(is_dev)
+    }
+
+    fn trie_cache_maximum_size(&self) -> Result<Option<usize>> {
+        self.base.base.trie_cache_maximum_size()
+    }
+
+    fn rpc_methods(&self) -> Result<sc_service::config::RpcMethods> {
+        self.base.base.rpc_methods()
+    }
+
+    fn rpc_max_connections(&self) -> Result<u32> {
+        self.base.base.rpc_max_connections()
+    }
+
+    fn rpc_cors(&self, is_dev: bool) -> Result<Option<Vec<String>>> {
+        self.base.base.rpc_cors(is_dev)
+    }
+
+    fn default_heap_pages(&self) -> Result<Option<u64>> {
+        self.base.base.default_heap_pages()
+    }
+
+    fn force_authoring(&self) -> Result<bool> {
+        self.base.base.force_authoring()
+    }
+
+    fn disable_grandpa(&self) -> Result<bool> {
+        self.base.base.disable_grandpa()
+    }
+
+    fn max_runtime_instances(&self) -> Result<Option<usize>> {
+        self.base.base.max_runtime_instances()
+    }
+
+    fn announce_block(&self) -> Result<bool> {
+        self.base.base.announce_block()
+    }
+
+    fn telemetry_endpoints(
+        &self,
+        chain_spec: &Box<dyn ChainSpec>,
+    ) -> Result<Option<sc_telemetry::TelemetryEndpoints>> {
+        self.base.base.telemetry_endpoints(chain_spec)
+    }
+
+    fn node_name(&self) -> Result<String> {
+        self.base.base.node_name()
+    }
+}
+
+impl SubstrateCli for RelayChainCli {
+    fn impl_version() -> String {
+        option_env!("SUBSTRATE_CLI_IMPL_VERSION")
+            .unwrap_or("Unknown version")
+            .into()
+    }
+
+    fn author() -> String {
+        option_env!("CARGO_PKG_AUTHORS")
+            .unwrap_or("Unknown authors")
+            .into()
+    }
+
+    fn support_url() -> String {
+        "https://github.com/paritytech/cumulus/issues/new".into()
+    }
+
+    fn copyright_start_year() -> i32 {
+        2020
+    }
+
+    fn load_spec(&self, id: &str) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
+        polkadot_cli::Cli::from_iter([Self::executable_name()].iter()).load_spec(id)
+    }
+
+    fn impl_name() -> String {
+        NODE_NAME.with(|n| format!("Container Chain {} Node", n.borrow()))
+    }
+
+    fn description() -> String {
+        NODE_NAME.with(|n| {
+            format!(
+                "Container Chain {} Node\n\nThe command-line arguments provided first will be \
+        passed to the parachain node, while the arguments provided after -- will be passed \
+        to the relay chain node.\n\n\
+        {} <parachain-args> -- <relay-chain-args>",
+                n.borrow(),
+                RelayChainCli::executable_name()
+            )
+        })
+    }
+}
+
+thread_local! {
+    static NODE_NAME: RefCell<String> = RefCell::new("Default".to_string());
```

Comment:
This will only work if `NODE_NAME.with()` is called from the same thread that run `set_node_name`. Maybe try using some of the OnceLock / OnceCell stuff from std instead

Reply:
Not a fan of globals like that. It would be cleaner for RelayChainCli to be generic over such constant, such as a Get<&'static str>.

Reply:
Yeah, since it's known at compile time it should be easy to turn into a generic, but if it's hard to do then I don't mind the global

Reply:
Will check guys. 
I was trying to find a way to override trait function implementation for the specific chain, but not sure if it's possible easily without overriding the whole trait which doesn't make sense. 
Will try to replace it with generic.

Reply:
Fixed

Reply:
@nanocryk, please also take a look. Thanks!

Reply:
Way better!